### ISSUE TITLE
Alt+Tab to Notepad++ sometimes the cursor/inputpoint is nowhere

### ISSUE THREAD

**Issue by libove** [notepad-plus-plus/notepad-plus-plus#9310]
### Description of the Issue
Starting .. recently? 7.9.1 update? Apologies, I don't know exactly, the following:
(Windows 10 20H2, 64-bit) Alt+Tab into Notepad++, the cursor/input point will be 'nowhere'.
Historically - and still today mostly - upon Alt+Tabbing into Notepad++ the cursor/input point will be wherever it most recently was in whatever buffer happens to be the current buffer.
However, for some weeks now, on multiple computers, upon Alt+Tabbing into Notepad++, occasionally, the cursor/input point will be 'nowhere'. That is, no keystroke will work. (I admit, I may not have tried Alt+ commands, but I HAVE hit ESC a couple of times to cancel possible unintended half Alt- keystrokes, and that hasn't worked, so it's not that a menu option has been partly activated). There is no blinking cursor. No menubar item is highlighted the way it would be if the user had simply tapped "Alt" for example.
It requires clicking with the mouse anywhere in Notepad++ to get the keyboard active again.

### Steps to Reproduce the Issue
1. (unreliably) Alt+Tab into Notepad++
2. Cursor may be 'nowhere'

### Expected Behavior
Cursor would always be live, somewhere in the Notepad++ window, with never the need to mouseclick before being able to start typing into the current buffer.

### Actual Behavior
Occasionally, unpredictably, the cursor is 'nowhere', and a mouseclick is required.

### Debug Information
Notepad++ v7.9.1   (64-bit)
Build time : Nov  2 2020 - 01:07:46
Path : C:\Program Files\Notepad++\notepad++.exe
Admin mode : OFF
Local Conf mode : OFF
OS Name : Windows 10 Pro (64-bit) 
OS Version : 2009
OS Build : 19042.685
Current ANSI codepage : 1252
Plugins : none

Hm. MAYBE it might occur after a buffer has updated because Notepad++ noticed that the underlying file had changed. This happened, once, as I was Alt+Tabbing quite a lot just to see if I could reproduce the problem, and in fact in a couple of dozen Alt+Tabs in and out of Notepad++ it did occur twice. I noticed in that particular case - Alt+Tab in, Notepad++ notices that an underlying file had changed and asks me if I want to update; I click yes and the cursor is nowhere until I Alt+Tab out and back in again. HOWEVER this is NOT the usual case - usually when I've encountered this issue in recent weeks, I did NOT also see a do-you-want-to-update prompt, the underlying file had not changed. (I do not have the option set to automatically reload buffers when underlying files change). I've continued lots of Alt+Tabbing and cannot reproduce the issue again just at the moment.

**Comment by zanud** [comment]
In my practice this problem frequently happens after switching from the "Find" (Ctrl-F) dialog to the main window. (File been editing is not changed externally.)

**Comment by libove** [comment]
Thank you @zanud . 
I tried to reproduce that, but in my case if the Find dialog is open then the Find dialog is the only thing to which I can Alt+Tab. That is, I cannot use Alt+Tab to switch between the (Notepad++) Find dialog and the (Notepad++) main window. Windows seems to treat the Notepad++ Find dialog as the only part of Notepad++ which is selectable by Alt+Tab. So I think that can't be exactly what I'm seeing. Regardless, it's good to hear that I'm not the only person experiencing it, even if we end of provoking it in different ways!

**Comment by sasumner** [comment]
Not sure I've ever noticed this happening to me, but I have heard of this "trick" as a workaround, that may ease the situation (or maybe not):

When it happens, try pressing the Windows key twice.

**Comment by zanud** [comment]
> That is, I cannot use Alt+Tab to switch between the (Notepad++) Find dialog and the (Notepad++) main window. Windows seems to treat the Notepad++ Find dialog as the only part of Notepad++ which is selectable by Alt+Tab.

Not exactly. If you are currently in the Find dialog, then **first** Alt+Tab will switch to the main Notepad++ window. And only after that any subsequent Alt+Tab will switch between Notepad++ and windows of other applications.

Right now in freshly launched Notepad++ I can not reproduce the problem. But during the last week I was actively working with a dozen of simultaneously opened files without closing the editor, and all this week the disappearing cursor (caret) drove me nuts (as I prefer to use keyboard only, without mouse).
So, perhaps, Ctrl-Tab also has some impact.

**Comment by libove** [comment]
Hi @zanud - that's simply not how it's working for me. If a Notepad++ Find dialog is open, that becomes "The" (only) Notepad++ window that Alt+Tab will see as being available to switch to. I did a quick search to see if Alt+Tab behavior is customisable, and found only size/transparency/position Registry settings, nothing that would seem to cause this difference in experience between our systems.

**Comment by dwahgon** [comment]
hi, i'm not sure if its entirely the same issue but i noticed notepad++ text cursor consistently losing focus when alt-tabbing into it out of a full-screen application, like a game. the text cursor would be where it should be for a second then un-focus. the notepad++ window itself is still "in focus" according to windows, but pressing anything does absolutely nothing. the suggestion above about pressing the windows key twice is a workaround that works and makes it focus back on the text, but is obviously not ideal. i've had this consistently for years now and im really surprised its not a more commonly talked about issue..

this also happens even when clicking into notepad++ out of a full-screen (or borderless) application, so it doesnt seem like the alt or tab keys are the culprit here, perhaps something relating to win11?

**Comment by xomx** [comment]
> losing focus when alt-tabbing into it out of a full-screen application

Cannot reproduce, I tried with:
https://github.com/gametutorials/tutorials/raw/refs/heads/master/Win32/Full%20Screen/FullScreen.exe
and then Alt-Tab to the N++ without such a problem with focus.

**Comment by dwahgon** [comment]
@xomx downloaded the FullScreen.exe on my side and you're right, this one doesnt make the issue appear, both when ran on its own and through steam

i went and tested it with a few more games and it seems to be specifically games running in windowed fullscreen, ones that cover the entire screen and allow easy alt-tabbing, but it is not reproducable with borderless windowed games that don't attempt to cover the entire screen, like Void Stranger

upon testing a bit further i can also reproduce it when switching a game (in this case Strange Antiquities Demo on steam) from borderless windowed to windowed, and upon switching the game window is still big enough to be dipping under the windows taskbar. tabbing out into notepad++ in this case makes the issue appear when windows makes an effort to hide the taskbar because it notices the game behind it, i guess. resizing the game window to something smaller so that it no longer encroaches the taskbar's personal space resolves the issue in this case

this is all with:
windows 11 24H2, 26120.4520, feature experience pack 1000.26100.154.0  (although it happened on non-insider builds as well in the past)
fresh install of notepad++ v8.8.3 64-bit, no plugins
